var/spack/repos/builtin/packages/celeritas/package.py

```diff
@@ -111,14 +118,20 @@ def cmake_args(self):
             from_variant("CELERITAS_BUILD_DOCS", "doc"),
             define("CELERITAS_BUILD_DEMOS", False),
             define("CELERITAS_BUILD_TESTS", False),
-            from_variant("Celeritas_USE_HIP", "rocm"),
+            from_variant("CELERITAS_USE_HIP", "rocm"),
             define("CELERITAS_USE_MPI", False),
             define("CELERITAS_USE_Python", True),
         ]
 
         for pkg in ["CUDA", "Geant4", "HepMC3", "OpenMP", "ROOT", "SWIG", "VecGeom"]:
             args.append(from_variant("CELERITAS_USE_" + pkg, pkg.lower()))
 
+        if self.spec.satisfies("+cuda"):
+            args.append(CMakeBuilder.define_cuda_architectures(self))
+        if self.spec.satisfies("+rocm"):
+            args.append(CMakeBuilder.define_hip_architectures(self))
+            args.append(define("Thrust_ROOT", self.spec["rocthrust"].prefix))
```

Comment:
This does not really make sense. The `find_package(Thrust)` is inside an `if(CELERITAS_USE_CUDA)`. I was confused by the reference to `rocthrust` inside `FindThrust.cmake`. I don't think there's a way to find `rocthrust` in the current celeritas build system.

Reply:
```suggestion
```

Reply:
Yes, you're correct, the `Thrust` package should only be set witih CUDA for now. I tried at one point adding `rocthrust` but it's not well behaved over ROCm versions (and indeed seems only to have shown up in 6.x). The relationship between it and CUDA, especially with version number reporting, is obtuse as well. And even the relationship between Thrust/CUDA and ROCm/Rocthrust is difficult (where does it live, when is it *not* installed as part of the dev resources, etc.)

It might make things easier on you (I do really appreciate all the work you did debugging this!) if we get together for a short zoom meeting sometime to hammer out any remaining confusion/issues in the spack package/Celeritas cmake?